core/string/ustring.cpp

```diff
@@ -4378,7 +4378,7 @@ String String::simplify_path() const {
 	}
 	Vector<String> dirs = s.split("/", false);
 
-	for (int i = 0; i < dirs.size(); i++) {
+	for (int i = 1; i < dirs.size(); i++) {
```

Comment:
This is wrong, you can see that the `for` loop explicitly handles `i == 0` below.

And a leading `./` needs to be removed.

What should be done here is remove the `i == 0` branch in `d == ".."`.

And adjust the tests accordingly as they seem to validate the wrong behavior.

Reply:
Done

Reply:
Yes you also need to fix that test.
```diff
diff --git a/tests/core/string/test_string.h b/tests/core/string/test_string.h
index cf57183a02..1b27966da1 100644
--- a/tests/core/string/test_string.h
+++ b/tests/core/string/test_string.h
@@ -1594,7 +1594,7 @@ TEST_CASE("[String] Path functions") {
 	static const char *base_name[8] = { "C:\\Godot\\project\\test", "/Godot/project/test", "../Godot/project/test", "Godot\\test", "C:\\test", "res://test", "user://test", "/" };
 	static const char *ext[8] = { "tscn", "xscn", "scn", "doc", "", "", "", "test" };
 	static const char *file[8] = { "test.tscn", "test.xscn", "test.scn", "test.doc", "test.", "test", "test", ".test" };
-	static const char *simplified[8] = { "C:/Godot/project/test.tscn", "/Godot/project/test.xscn", "Godot/project/test.scn", "Godot/test.doc", "C:/test.", "res://test", "user://test", "/.test" };
+	static const char *simplified[8] = { "C:/Godot/project/test.tscn", "/Godot/project/test.xscn", "../Godot/project/test.scn", "Godot/test.doc", "C:/test.", "res://test", "user://test", "/.test" };
 	static const bool abs[8] = { true, true, false, false, true, true, true, true };
 
 	for (int i = 0; i < 8; i++) {
```

Reply:
Thanks!